src/crystal/system/unix/env.cr

```diff
@@ -0,0 +1,55 @@
+require "c/stdlib"
+
+module Crystal::System::Env
+  # Sets an environment variable.
+  def self.set(key : String, value : String) : Nil
+    raise ArgumentError.new("Key contains null byte") if key.byte_index(0)
```

Comment:
`key.check_no_bull_byte`

Reply:
That's what I initially thought, too. But `check_no_null_byte` loses the information which of the arguments was invalid (it's `String contains null byte` instead of `Key contains null byte`). Doesn't make much difference, so I'm not sure about it...

Reply:
Then refactor it to take an argument